### DISCUSSION TITLE
Issue with Branch Handling in JReleaser on GitLab

### DISCUSSION THREAD

**Discussion by hamzasalaheddine** [jreleaser/jreleaser#1718] [Q&A] [answered]
Hello,

I am currently using JReleaser version 1.14.0 and have encountered an issue related to the handling of branches in the release configuration on GITLAB

I notice that JReleaser automatically appends  value of branch name @master to the end of the repository URL

For example, the URL being used is:

https://{{repoHost}}/{{repoOwner}}/{{repoName}}@master

I have ensured that the repository URL is correct and that the token used has the necessary permissions. 

Despite these configurations, the @master suffix continues to be added to the URL, resulting in a 404: Project does not exist error.

[ERROR] JReleaser failed after 1.035 s
Unexpected error when creating release
org.jreleaser.sdk.commons.RestAPIException: 404: Project {{repoName}} does not exist or it's not visible

normally URL with branch name should be

https://{{repoHost}}/{{repoOwner}}/{{repoName}}/-/tree/master NOT https://{{repoHost}}/{{repoOwner}}/{{repoName}}@master

Could you please advise on how to resolve this issue or provide any guidance on how to prevent JReleaser from appending the branch to the URL? 

Any help or suggestions would be greatly appreciated.

Thank you for your assistance.

Best regards,
Hamza

**Comment by aalmiray** [comment]
Could you post the `release` section of your configuration? i'm assuming you may be using the `jreleaser.yml` file. Also, please run `jreleaser config` and post the `release` section found in `out/jreleaser/trace.log`.

**Comment by hamzasalaheddine** [comment]
[jreleaser-trace.txt](https://github.com/user-attachments/files/16841452/jreleaser-trace.txt)
Thank you for your prompt response.

To provide some context, we are using JReleaser to automate the release creation on GitLab when a tag is created.

We have a GitLab Enterprise instance on the cloud (unfortunately, I cannot share the exact link).

Here is the release section in the jreleaser.yml file:

release:
  gitlab:
    owner: mario
    host: gitlab.sh.interne.cloud.xxx.com
    apiEndpoint: http://gitlab.sh.interne.cloud.xxx.com
    branch: master
    repoUrl: http://gitlab.sh.interne.cloud.xxx.com/mario/euro-api

In the .gitlab-ci.yml, I have the release stage configured as follows:

release:
  stage: release
  image: jreleaser/jreleaser-alpine:v1.14.0
  script:
    - echo "Running JReleaser for tag $CI_COMMIT_TAG"
    - jreleaser full-release -Djreleaser.project.version="$CI_COMMIT_TAG" -Djreleaser.gitlab.token="MARIO_CI_TOKEN"
  only:
    - tags

Please find attached the configuration generated in out/jreleaser/.log.

Thank you for your help and availability.

**Reply by aalmiray** [reply, marked as answer]
Thank you. Have you tried setting the `projectIdentifier` property in the `release/gitlab` section?

**Reply by hamzasalaheddine** [reply]
Thank you very much! That was exactly what was missing. It’s working perfectly now. I appreciate your quick response and your helpful assistance!
